shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -677,6 +677,15 @@ public MotionEvent toMotionEvent(
         MotionEventTracker.MotionEventId.from(touch.motionEventId);
     MotionEvent trackedEvent = motionEventTracker.pop(motionEventId);
 
+    if (!usingVirtualDiplay && trackedEvent != null) {
+      // We have the original event, deliver it as it will pass the verifiable
+      // input check.
+      return trackedEvent;
+    }
+    // We are in virtual display mode or don't have a reference to the original MotionEvent.
+    // In this case we manually recreate a MotionEvent to be delivered. This MotionEvent
+    // will fail the verifiable input check.
+
```

Comment:
What are the cases are where we don't have the original event if virtual display isn't being used (if we know them)? And why won't we have it when virtual display is used? Do these cases warrant warning users here?

Reply:
IIUC, the only case where this will happen (when not using virtual display) is when a gesture detector runs and creates new events related to the detected gesture.